src/file-upload/dropzone.tsx

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './styles.css.js';
+import clsx from 'clsx';
+import InternalIcon from '../icon/internal';
+import InternalBox from '../box/internal';
+
+interface DropzoneProps {
+  onChange: (files: File[]) => void;
+  children: React.ReactNode;
+}
+
+export function useDropzoneVisible() {
+  const [isDropzoneVisible, setDropzoneVisible] = useState(false);
+
+  // Registering global drag events listeners.
+  useEffect(() => {
+    // The timer helps avoiding dropzone blinking.
```

Comment:
Do we know why the blinking happens?

Reply:
Did you consider using `dragenter` event instead? 

https://html.spec.whatwg.org/multipage/dnd.html#event-dnd-dragenter

That one seem to be strictly attached to the document body

Reply:
I think the main problem is not with dragover but with dragleave. If there was a reliable event to be used instead that could have helped. The dragleave is issued whenever cursor leaves UI elements and it needs to be cancelled by the dragover.

Reply:
You can check the event target to figure which element was left. Try to test locally, maybe `event.target.tagName === 'BODY'` will work

Reply:
Good idea - I'll try it out 👍

Reply:
The dragleave event's target is not reliable. It can be html, main or some div when the element is dragged away from the page.

Reply:
Timeouts are even more unreliable. We had similar issue in dropdown component: https://github.com/cloudscape-design/components/pull/541

Let's try harder and avoid timeouts. DM me if you are very stuck